spanner/google/cloud/spanner_v1/keyset.py

```diff
@@ -38,14 +38,15 @@ class KeyRange(object):
     """
     def __init__(self, start_open=None, start_closed=None,
                  end_open=None, end_closed=None):
-        if not any([start_open, start_closed, end_open, end_closed]):
-            raise ValueError("Must specify at least a start or end row.")
-
-        if start_open and start_closed:
+        if start_open is not None and start_closed is not None:
             raise ValueError("Specify one of 'start_open' / 'start_closed'.")
+        elif start_open is None and start_closed is None:
+            start_closed = []
```

Comment:
I think this should be an exception. If not, you need to update the docstring.

Reply:
this has to be there or you can't do 
KeyRange(end_open = x)

Reply:
But this also makes the empty constructor allowed.  I can add back in the (if not any...)

Reply:
So here is what I'm saying, we have two options:

1. Instead of setting `start_closed = []` here, we raise an exception stating that exactly one of `start_open` and `start_closed` must be provided
1. We must update the `start_open /  start_closed` docstrings to explain that if **neither** is provided, then we implicitly take it to mean `start_closed=[]`

Similar applies to `end_*`. My comment above (about the need for the `all()` check) indicates my preference (which is getting stronger by the minute) for the "raise an exception" choice.

Reply:
1 does make it more consistent with the behavior of the empty constructor.